### PR TITLE
Bux fix: A stroke consisting of two equal crashes xournal

### PR DESCRIPTION
**Pull request by immi-h**

### CODE REVIEW

File: src/xo-file.c [hunk 1]
```diff
@@ -978,7 +978,8 @@ void xoj_parser_text(GMarkupParseContext *context,
     /* Note: comparing floats with "==" is dangerous, though in the case of dummy points, both floats
        will have been computed in the same way from the same ascii string, so they should really be
        absolutely equal. */
-    if (ui.cur_path.coords[0] == ui.cur_path.coords[2] && ui.cur_path.coords[1] == ui.cur_path.coords[3]) {
+    if (ui.cur_path.num_points >= 3 &&
```
Comment (dmgerman) [inline src/xo-file.c]: If the number of points is 2 and the points are different then you drop the first point? even if they are different?

**Review by dmgerman — COMMENTED** [review]
Please see my comment regarding the dropping of the first stroke. My suggestion is

if (first and second points are the same) then drop first point
otherwise leave the same.

That logic is simpler.

The current version is:

if n> 3 and points are the same...

**Comment by immi-h** [comment]
I don't exactly understand your suggestion. (Maybe I'm misunderstanding it.) It seems to me that what you're suggesting is exactly the same as before the bugfix.
The version before the bugfix was "if (first and second points are the same) then drop first point"
If there are exactly two equal points, this leaves only one point, which crashes xournal. Therefore, I added the additional condition "if n >= 3".

**Comment by immi-h** [comment]
New change: Don't drop a point of fixed-width strokes. (This could also cause a crash, and we don't want to drop such points anyway.)

Note: It might seem that already the prev. version fixed everything, but probably not in a reliable way: For fixed-width storkes, only n contains the number of points (and not ui.cur_path.num_points, which probably is uninitialized), so if ui.cur_path.num_points happens to be 0 for fixed-width strokes, everything seemed to work..

All this is fixed now.

**Pull request merged**